go/disk.go

```diff
@@ -0,0 +1,352 @@
+package hyperkit
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+const (
+	mib = int64(1024 * 1024)
+)
+
+/*-------.
+| Disk.  |
+`-------*/
+
+// Disk in an interface for qcow2 and raw disk images.
+type Disk interface {
+	// GetPath returns the location of the disk image file.
+	GetPath() string
+	// SetPath changes the location of the disk image file.
+	SetPath(p string)
+	// GetSize returns the desired size of the disk image file.
+	GetSize() int
+	// String returns the path.
+	String() string
+
+	// Exists is true iff the disk image file exists.
+	Exists() bool
+	// Ensure creates the disk image if needed, and resizes it if needed.
+	Ensure() error
+	// Stop can be called when hyperkit has quit.  It performs sanity checks, compaction, etc.
+	// It can be passed a lock file which is not-used, but kept alive even if the parent
+	// process died.
+	Stop(lockFile *os.File) error
+
+	// AsArgument returns the command-line option to pass after `-s <slot>:0,` to hyperkit for this disk.
+	AsArgument() string
+
+	create() error
+	getFileSize() (int, error)
+	resize() error
+}
+
+// exists if the image file exists.
+func exists(d Disk) bool {
+	_, err := os.Stat(d.GetPath())
+	return err == nil
```

Comment:
`os.Stat` can return errors other than `os.ErrNotExist`.

Should either log in that case or return the `err` for the caller to deal with, which would be something like:
```
if err== nil || os.IsNotExist(err) {
    return err == nil, nil
}
return false, err
```
Perhaps that can be simplified though.

I can't actually find any callers of the `Exists` functions which wrap this in the PR though so I'm not sure which would be more appropriate. Maybe this could even just be removed if it is unused.

Reply:
Hi Ian,

Yes, I am well aware of the different types of error.  But this is the code as it lived in Piñata, and over there, what mattered is whether the file existed.  Errors counted as "does not exist", and then the code that deals with the non-existing situation would deal with finer grained errors.

But I can definitely add logs here.